spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/support/AbstractRibbonCommand.java

```diff
@@ -74,9 +84,11 @@ protected static Setter getSetter(final String commandKey) {
 	@Override
 	protected ClientHttpResponse run() throws Exception {
 		final RequestContext context = RequestContext.getCurrentContext();
-		String contentLengthHeader = context.getRequest().getHeader("Content-Length");
-		if (StringUtils.hasText(contentLengthHeader)) {
-			this.context.setContentLength(new Long(contentLengthHeader));
+		if (context.getRequest() != null) {
```

Comment:
However, I think this whole code is not necessary as RibbonCommandContext constructor has the headers.
Content Length should probably be retrieved at this point.